### PR TITLE
Add support for dotnet isolated apps in azure static web

### PR DESCRIPTION
**Pull request by arroyc**
- [X] The purpose of this PR is explained in this message or in an issue. 
- [X] Tests are included and/or updated for code changes.
- [X] Proper license headers are included in each file.

### CODE REVIEW

**Comment by arroyc** [comment]
will add/update necessary tests shortly

File: images/build/Dockerfiles/azureFunctions.JamStack.Dockerfile [hunk 1]
```diff
@@ -26,4 +28,9 @@ RUN oryx prep --skip-detection --platforms-and-versions nodejs=12 \
     && downloadedFileName="go${GO_VERSION}.linux-amd64.tar.gz" \
     && curl -SLsO https://golang.org/dl/$downloadedFileName \
     && tar -C /usr/local -xzf $downloadedFileName \
-    && rm -rf $downloadedFileName
+    && rm -rf $downloadedFileName \ 
+    && echo "jamstack" > /opt/oryx/.imagetype
+    
+RUN ./opt/tmp/build/createSymlinksForDotnet.sh \
+    && ls -la /home/jamstack/ \
```
Comment (william-msft) [inline images/build/Dockerfiles/azureFunctions.JamStack.Dockerfile:35]: Removing debugging print lines

File: src/BuildScriptGenerator/DefaultBuildScriptGenerator.cs [hunk 2]
```diff
@@ -80,7 +80,18 @@ internal class DefaultBuildScriptGenerator : IBuildScriptGenerator
             {
                 foreach (var toolNameAndVersion in toolsToBeSetInPath)
                 {
-                    toolsToVersion[toolNameAndVersion.Key] = toolNameAndVersion.Value;
+                    _logger.LogInformation($"If {toolNameAndVersion.Key} is set as environment, it'll be not be set via benv");
+                    if (!string.IsNullOrEmpty(
+                        Environment.GetEnvironmentVariable(toolNameAndVersion.Key)))
+                    {
+                        _logger.LogInformation($"If {toolNameAndVersion.Key} is set as environment, it'll be not be set via benv");
+
```
Comment (william-msft) [inline src/BuildScriptGenerator/DefaultBuildScriptGenerator.cs:88]: Please remove log statement if not super relevant

File: images/build/Dockerfiles/gitHubActions.Dockerfile [hunk 3]
```diff
@@ -133,7 +133,7 @@ RUN tmpDir="/opt/tmp" \
         libonig-dev \
     && rm -rf /var/lib/apt/lists/* \
     && rm -f /etc/apt/sources.list.d/buster.list \
-    && rm -rf $tmpDir \
+    #&& rm -rf $tmpDir \
```
Comment (william-msft) [inline images/build/Dockerfiles/gitHubActions.Dockerfile:136]: Remove this line please

**Review by william-msft — APPROVED** [review]

**Pull request merged**